### PR TITLE
Additional daemon commands

### PR DESCRIPTION
**Pull request by brandonlehmann**
#### Additional Daemon Commands

* print_block_raw
* print_block_pow

### CODE REVIEW

File: src/Daemon/DaemonCommandsHandler.cpp [hunk 1]
```diff
@@ -251,6 +253,68 @@ bool DaemonCommandsHandler::print_block(const std::vector<std::string> &args) {
...
+}
+//--------------------------------------------------------------------------------
+bool DaemonCommandsHandler::print_block_raw_by_height(uint32_t height)
+{
+  if (height - 1 > m_core.getTopBlockIndex()) {
+    std::cout << "block wasn't found. Current block chain height: " << m_core.getTopBlockIndex() + 1 << ", requested: " << height << std::endl;
+    return false;
+  }
+
+  auto hash = m_core.getBlockHashByIndex(height - 1);
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.cpp]:
```suggestion
  const auto hash = m_core.getBlockHashByIndex(height - 1);
```

File: src/Daemon/DaemonCommandsHandler.cpp [hunk 2]
```diff
@@ -251,6 +253,68 @@ bool DaemonCommandsHandler::print_block(const std::vector<std::string> &args) {
...
+    uint32_t height = boost::lexical_cast<uint32_t>(arg);
+    print_block_raw_by_height(height);
+  } catch (boost::bad_lexical_cast &) {
+    std::cout << "Must supply a height" << ENDL;
+  }
+
+  return true;
+}
+//--------------------------------------------------------------------------------
+bool DaemonCommandsHandler::print_block_raw_by_height(uint32_t height)
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.cpp]:
```suggestion
bool DaemonCommandsHandler::print_block_raw_by_height(const uint32_t height)
```

File: src/Daemon/DaemonCommandsHandler.cpp [hunk 3]
```diff
@@ -251,6 +253,68 @@ bool DaemonCommandsHandler::print_block(const std::vector<std::string> &args) {
...
+{
+  if (height - 1 > m_core.getTopBlockIndex()) {
+    std::cout << "block wasn't found. Current block chain height: " << m_core.getTopBlockIndex() + 1 << ", requested: " << height << std::endl;
+    return false;
+  }
+
+  auto hash = m_core.getBlockHashByIndex(height - 1);
+  std::cout << ENDL << "block_id: " << hash << ENDL;
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.cpp]:
```suggestion
  std::cout << "\nblock_id: " << hash << std::endl;
```

File: src/Daemon/DaemonCommandsHandler.cpp [hunk 4]
```diff
@@ -251,6 +253,68 @@ bool DaemonCommandsHandler::print_block(const std::vector<std::string> &args) {
...
+  std::cout << ENDL << m_core.getRawBlockHex(height - 1) << ENDL;
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.cpp]:
```suggestion
  std::cout << "\n" << m_core.getRawBlockHex(height - 1) << std::endl;
```

File: src/Daemon/DaemonCommandsHandler.cpp [hunk 5]
```diff
@@ -251,6 +253,68 @@ bool DaemonCommandsHandler::print_block(const std::vector<std::string> &args) {
...
+//--------------------------------------------------------------------------------
+bool DaemonCommandsHandler::print_block_pow(const std::vector<std::string> &args) {
+  if (args.empty()) {
+    std::cout << "expected: print_block_pow <block_height>" << std::endl;
+    return true;
+  }
+
+  const std::string &arg = args.front();
+  try {
+    uint32_t height = boost::lexical_cast<uint32_t>(arg);
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.cpp]: Same comments as before apply

File: src/Daemon/DaemonCommandsHandler.cpp [hunk 6]
```diff
@@ -251,6 +253,68 @@ bool DaemonCommandsHandler::print_block(const std::vector<std::string> &args) {
...
+//--------------------------------------------------------------------------------
+bool DaemonCommandsHandler::print_block_pow_by_height(uint32_t height)
+{
+  if (height - 1 > m_core.getTopBlockIndex()) {
+    std::cout << "block wasn't found. Current block chain height: " << m_core.getTopBlockIndex() + 1 << ", requested: " << height << std::endl;
+    return false;
+  }
+
+  auto hash = m_core.getBlockHashByIndex(height - 1);
+  std::cout << ENDL << "block_id: " << hash << ENDL;
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.cpp]:
```suggestion
  std::cout << "\nblock_id: " << hash << std::endl;
```

File: src/Daemon/DaemonCommandsHandler.cpp [hunk 7]
```diff
@@ -251,6 +253,68 @@ bool DaemonCommandsHandler::print_block(const std::vector<std::string> &args) {
...
+  std::cout << ENDL << m_core.getBlockPoWHash(height - 1) << ENDL;
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.cpp]:
```suggestion
  std::cout << "\n" << m_core.getBlockPoWHash(height - 1) << std::endl;
```

File: src/Daemon/DaemonCommandsHandler.h [hunk 8]
```diff
@@ -45,7 +45,9 @@ class DaemonCommandsHandler
 
   std::string get_commands_str();
   bool print_block_by_height(uint32_t height);
+  bool print_block_raw_by_height(uint32_t height);
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.h]:
```suggestion
  bool print_block_raw_by_height(const uint32_t height);
```

File: src/Daemon/DaemonCommandsHandler.h [hunk 9]
```diff
@@ -45,7 +45,9 @@ class DaemonCommandsHandler
...
   bool print_block_by_hash(const std::string& arg);
+  bool print_block_pow_by_height(uint32_t height);
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.h]:
```suggestion
  bool print_block_pow_by_height(const uint32_t height);
```

File: src/Daemon/DaemonCommandsHandler.cpp [hunk 10]
```diff
@@ -251,6 +253,68 @@ bool DaemonCommandsHandler::print_block(const std::vector<std::string> &args) {
...
+    uint32_t height = boost::lexical_cast<uint32_t>(arg);
+    print_block_pow_by_height(height);
+  } catch (boost::bad_lexical_cast &) {
+    std::cout << "Must supply a height" << ENDL;
+  }
+
+  return true;
+}
+//--------------------------------------------------------------------------------
+bool DaemonCommandsHandler::print_block_pow_by_height(uint32_t height)
```
Comment (zpalmtree) [inline src/Daemon/DaemonCommandsHandler.cpp]:
```suggestion
bool DaemonCommandsHandler::print_block_pow_by_height(const uint32_t height)
```

File: src/CryptoNoteCore/Core.cpp [hunk 11]
```diff
@@ -3027,5 +3027,30 @@ std::time_t Core::getStartTime() const
...
+    {
+        return "";
+    }
+
+    RawBlock _block = mainChainStorage->getBlockByIndex(blockheight);
+
+    return Common::toHex(_block.block);
+}
+
+std::string Core::getBlockPoWHash(const uint32_t blockheight)
```
Comment (zpalmtree) [inline src/CryptoNoteCore/Core.cpp]: The naming of this suggests it's returning a `Crypto::Hash` to me. Maybe update it to return that, or rename it?

File: src/Daemon/DaemonCommandsHandler.cpp [hunk 12]
```diff
@@ -251,6 +253,68 @@ bool DaemonCommandsHandler::print_block(const std::vector<std::string> &args) {
...
 //--------------------------------------------------------------------------------
+bool DaemonCommandsHandler::print_block_raw(const std::vector<std::string> &args) {
+  if (args.empty()) {
+    std::cout << "expected: print_block_raw <block_height>" << std::endl;
+    return true;
+  }
+
+  const std::string &arg = args.front();
+  try {
+    uint32_t height = boost::lexical_cast<uint32_t>(arg);
```
[the remark this answers is not in the document]
Comment (brandonlehmann) [reply src/Daemon/DaemonCommandsHandler.cpp]: Copy pasta. It's not worth it.

**Comment by brandonlehmann** [comment]
Changed my mind. I found it useful but that's too much stuff to change for a helpful debugging tool.

**Pull request closed without merging**